integration-tests/client-server/src/main/java/io/quarkiverse/cxf/it/redirect/RedirectRest.java

```diff
@@ -93,16 +117,25 @@ public Response tripleRedirect() {
         return Response.status(302).header("Location", "/RedirectRest/doubleRedirect").build();
     }
 
-    @Path("/selfRedirect")
+    @Path("/selfRedirect/{selfRedirectsCount}/{resetCount}")
     @POST
-    public Response selfRedirect(@Context HttpHeaders headers) {
+    public Response selfRedirect(
+            @PathParam("selfRedirectsCount") int selfRedirectsCount,
+            @PathParam("resetCount") int resetCount) {
         int count = redirectCounter.incrementAndGet();
-        if (count <= NUM_SELF_REDIRECTS) {
-            return Response.status(302).header("Location", "/RedirectRest/selfRedirect").build();
+
+        if (selfRedirectsCount == resetCount && count == resetCount) {
+            redirectCounter.set(0);
```

Comment:
This is a smart way to do the counting, but one remaining issue is it doesn't really reset the counter to 0 after a test for too many same URIs. You can see that with the current setup with async/maxSameUri2, the counter begins at 1 instead of 0 due to the previous sync test. I don't have a good solution for this (other than to add yet another path param that copies the max same uri property), but would perhaps set the counter to -1 instead of 0 here to at least avoid this problem in this setup in case future tests are added.

Reply:
Oh, really, thanks for finding out. We could introduce a separate rest endpoint to reset the counter.

Reply:
Do you mean a separate endpoint only for the too many same uri test, which basically doesn't keep a counter and just keeps redirecting to self? So this endpoint is only kept for the tests with max-uri > number of self redirects, which means it always resets to 0 after?

Reply:
I looked at it again after a night's sleep. I think a new endpoint is not needed. If you change the URI of the maxSameURi2 test to selfRedirect/3/3 it should solve the issue, which makes sense because the selfRedirectCount indicates how many self redirects there should be before the test completes (either going to a real endpoint or throwing an error), and for maxSameUri2 we need 3 self redirects before an error is thrown. 

The condition can also be simplified to 
`
if (count == resetCount) {
    redirectCounter.set(0);
}
`

Reply:
See 7bb1c1b where I added `RedirectRest.selfRedirect()` for resetting the counter.

Reply:
I never thought about it that way, interesting solution. Question: why do the try-finally in the redirect test instead of splitting it into two tests (small/large body) and let the BeforeEach method do the job?

Reply:
No specific reason, could be done that way too.